tests/unit/layers/models/fileValidator.spec.ts

```
@@ -46,49 +47,166 @@ describe('FileValidator', () => {
     });
   });
 
-  describe('validateProjections', () => {
-    it('should not throw an error due to valid projection', async function () {
-      getProjectionMock.mockReturnValue('4326');
+  describe('validateGpkgFiles', () => {
+    it('should fail if geopackage does not have a tiles index', function () {
+      const getGpkgIndexFn = jest.spyOn(SQLiteClient.prototype as unknown as { getGpkgIndex: () => unknown }, 'getGpkgIndex');
+      getGpkgIndexFn.mockReturnValue(undefined);
+      setValue({ layerSourceDir: 'tests/mocks' });
+      const testData: string[] = ['unindexed.gpkg'];
+      const fileValidator = new FileValidator(configMock, jsLogger({ enabled: false }), gdalUtilitiesMock);
+
+      const action = () => fileValidator.validateGpkgIndex(testData, 'files');
+      expect(action).toThrow(BadRequestError);
+      expect(getGpkgIndexFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('should not throw an error if geopackage have a tile index', function () {
+      const getGpkgIndexFn = jest.spyOn(SQLiteClient.prototype as unknown as { getGpkgIndex: () => unknown }, 'getGpkgIndex');
+      getGpkgIndexFn.mockReturnValue({});
       setValue({ layerSourceDir: 'tests/mocks' });
       const testData: string[] = ['indexed.gpkg'];
       const fileValidator = new FileValidator(configMock, jsLogger({ enabled: false }), gdalUtilitiesMock);
 
-      const action = async () => fileValidator.validateProjections(testData, 'files');
+      const action = () => fileValidator.validateGpkgIndex(testData, 'files');
+      expect(action).not.toThrow();
+      expect(getGpkgIndexFn).toHaveBeenCalledTimes(1);
+    });
+
+    it('should return true if geopackage file is correct', function () {
```

add test of should fail for invalid "SQLlite" file